### DISCUSSION TITLE
The time "Running queries for ruby" increased by syntax_suggest

### DISCUSSION THREAD

**Discussion by nobu** [github/codeql#10120] [Q&A]
Recently, ruby master incorporated [`syntax_suggest`](https://github.com/ruby/ruby/commit/490af8dbdb66263f29d0b4e43752fbb298b94862), then the time "Running queries for ruby" increased one hour.
Before that commit, it was about a [quarter-hour](https://github.com/ruby/ruby/runs/7909936208?check_suite_focus=true#step:9:1496).

```
[32/32 eval 16m4s] Evaluation done; writing results to codeql/ruby-queries/queries/security/cwe-1333/ReDoS.bqrs.
```

But now it is [plus an hour](https://github.com/ruby/ruby/runs/7910040804?check_suite_focus=true#step:9:1496).

```
[32/32 eval 75m46s] Evaluation done; writing results to codeql/ruby-queries/queries/security/cwe-1333/ReDoS.bqrs.
```

From the metric data, the difference of total lines doesn't seem significant, comparing with the above times.

[Before](https://github.com/ruby/ruby/runs/7909936208?check_suite_focus=true#step:9:1904)
|                         Metric                          |  Value  |
|---------------------------------------------------------|---------|
| Total lines of Ruby code in the database                | 1240764 |
| Total Lines of user written Ruby code in the database   | 1232069 |
| Total number of files that were extracted with errors   |      29 |
| Total number of files that were extracted without error |    7914 |

[After](https://github.com/ruby/ruby/runs/7910040804?check_suite_focus=true#step:9:1904)
|                         Metric                          |  Value  |
|---------------------------------------------------------|---------|
| Total lines of Ruby code in the database                | 1242280 |
| Total Lines of user written Ruby code in the database   | 1233585 |
| Total number of files that were extracted with errors   |      29 |
| Total number of files that were extracted without error |    7940 |

Any suggestions are appreciated.
Thanks in advance.

**Comment by nickrolfe** [comment]
Hi @nobu, thanks for bringing this to our attention.

I am able to reproduce the performance problem running our queries against your `master` branch. We will investigate and aim to fix this.

**Reply by aibaars** [reply]
@noby as a temporary workaround you could try adding the following snippet to [`.github/codeql/codeql-config.yml`](https://github.com/ruby/ruby/blob/master/.github/codeql/codeql-config.yml). That should make CodeQL ignore the new folder. 
```
paths-ignore:
  - lib/syntax_suggest
```

See also: [Specifying directories to scan](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-code-scanning#specifying-directories-to-scan)

**Reply by nobu** [reply]
Thank you for the info.
Applied it now.

**Reply by aibaars** [reply]
@nobu I'm happy to see that it worked for you. Note that the next version of CodeQL 2.10.5 should have a fix for this problem, so you might want to revert the workaround in a week or two.

**Reply by nobu** [reply]
Thank you, reverted the filter as all instances seem updated.

**Reply by aibaars** [reply]
Great! Thanks for confirming.
